Fix class name in ProductDownloadsServiceProvider

szepeviktor opened the pull request:
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fix class name

From https://github.com/woocommerce/woocommerce/pull/37015#issuecomment-1453436541

szepeviktor committed:
Fix class name in ProductDownloadsServiceProvider

barryhughes committed:
Add changelog file

szepeviktor commented:
@barryhughes Thank you for the changelog entry.
If you get me €30/hour I could keep on sending these tech debt type PR-s for a very-very long time.
(with my laser eyes)

barryhughes commented:
If [Peter can't do so](https://github.com/woocommerce/woocommerce/pull/37015#issuecomment-1453615225), I'm afraid I have even less chance—we certainly appreciate the contributions and your laser eyes, though ... and we should perhaps also integrate relevant tool(s) like PhpStan into our workflows.

szepeviktor commented:
All right, all right!
Cheap labor is a no go.

barryhughes committed:
Changelog file added to the wrong location.

barryhughes committed:
Add changelog file

barryhughes committed:
Add changelog file

barryhughes committed:
Remove incorrectly-located changelog file

(Alternative changelogger tool for community PRs was writing to the wrong location.)
